### DISCUSSION TITLE
Add HA Player to MA

### DISCUSSION THREAD

**Discussion by collateral87** [music-assistant/support#1918] [Q&A] [answered]
Hi everyone,
i need some support to add a Homeassistant Player to Musicassistant via the Plugin. When i select a player, the spinning loading animation comes, but nothing happens.

![grafik](https://github.com/music-assistant/hass-music-assistant/assets/97076257/4fdd3b52-98b3-43d4-b818-35c9bba9d89e)
my plugin configuration

![grafik](https://github.com/music-assistant/hass-music-assistant/assets/97076257/ad64f458-854c-4ab7-80d2-dd36b2de095b)
screenshot of the issue

Do i miss something?

Thank you

**Comment by Smarthome-Creator** [comment]
@collateral87 , I have a question about the setup, as I can't find any documentation anywhere. The URL cannot be accessed, I have tried it in more than one browser.

Or to put it another way, which URL should be included? Once I have overcome this hurdle, perhaps I can test it.

Greetings

**Reply by collateral87** [reply]
The URL http://supervisor/core/api was already entered in the field for me. I neither know if this is correct nor what else I should enter here. Also, regarding the token, I'm not sure what kind of token is needed here.

Maybe someone from the team can provide more information. I believe this is the best and most powerful feature in Music Assistant, and I would love to try it out.

**Reply by Smarthome-Creator** [reply]
After I simply clicked on "Save", the HA Media Player works for me without any problems.
If I were you, I would restart the AddOn or HA completely.
And then just try again.

**Reply by collateral87** [reply]
can you import mediaplayer as well? or are you talking only about the setup of the plugin, which works well?

i will try a complete reboot now and will report the result.

**Reply by collateral87** [reply]
Bad same issue. When i click save on import player window, the loading animation goes forever :(

**Reply by OzGav** [reply]
Hopefully this is fixed in the latest beta.

**Reply by collateral87** [reply]
The issue is solved indeed. Thank you :)

But i testet every single HA player i have. None of them worked (connection failed in the logs).
I saw screenshots of Imported browsermod players, but they dont work too.
I think i test everything and it feels like, that only i have this problem.

**Reply by OzGav** [reply]
Its possibly a network problem then. Are you doing anything non-standard. Firewalls. VPN, proxies, mutiple subnets, VLANs etc

**Reply by marcelveldt** [reply]
Have a look in the logs after trying to use the HA players in MA and see what it says.

**Reply by collateral87** [reply]
I appreciate your help. Thank you.
My setup is HassOS in a VM on an Unraid server. MA is running as addon in HA. My router is a FritzBox. I think that my settings are relatively standard. I'm not aware of any strange network settings. I have also not seen such a phenomenon since I started using Home Assistant. I have not had any network problems at all.

![grafik](https://github.com/music-assistant/hass-music-assistant/assets/97076257/79a79697-13a3-4a90-b03f-0e6039355a3d)

**Reply by collateral87** [reply]
can you answer me this questions please:

- are there any settings i can play with?
- or do you know any tips or tricks how i can isolate the error?
- when all other player provider (including fully kiosk) works flawless, indicates this still a network error, when only the ha-provider give me this error?

thank you

**Reply by OzGav** [reply]
Have you tried removing the integration, removing from HACS, restarting HA and then reinstallng the integration?

**Reply by collateral87** [reply]
I testet this again now and sadly no changes.

I didnt know, that the ha player provider relies on the intgration.
Do you have any other ideas that i can try?

**Reply by OzGav** [reply]
I’m looking at your initial post. Can you confirm what you have entered in the URL field? It should be your HA IP

**Reply by collateral87** [reply]
Since beta 98+ there isnt such a field anymore.

It looks like:

![Screenshot_20240226_235300_Home Assistant.jpg](https://github.com/music-assistant/hass-music-assistant/assets/97076257/16519128-fb91-4672-b0e8-187870164f10)

And in the Integration setup i entered my ip + 8095.

**Reply by OzGav** [reply]
OK in the Integration setup if you are using HAOS this should be auto populated with the docker IP (I think ...mine is). Can you show a screenshot when you press configure on the Integration in HA settings...

When you set it up you should just leave the top box ticked if using HAOS

![image](https://github.com/music-assistant/hass-music-assistant/assets/19848947/420a31c7-ae5e-48f5-bb3f-d0d4f670351e)

**Comment by collateral87** [comment]
![grafik](https://github.com/music-assistant/hass-music-assistant/assets/97076257/9e3698c6-82bd-4504-8b49-4cd2eb795fd8)
![grafik](https://github.com/music-assistant/hass-music-assistant/assets/97076257/b9957058-4c54-4b5f-b38d-6bb358ac495e)

**Comment by collateral87** [comment]
![grafik](https://github.com/music-assistant/hass-music-assistant/assets/97076257/aa103e22-c854-4241-8429-9bc9cb112077)

**Reply by OzGav** [reply]
go to MA SETTINGS>>CORE and download the log file. Post the top of the file down to and including the line that looks like:
`2024-02-27 10:07:22.429 INFO (MainThread) [music_assistant.webserver] Starting server on  172.30.32.1:8095 - base url: http://172.30.32.1:8095`

**Comment by collateral87** [comment]
2024-02-27 01:12:25.036 INFO (MainThread) [music_assistant] Starting Music Assistant Server (2dec3a49b56a4eef85db734baee7ecbf) version 2.0.0b106 - uvloop: False
2024-02-27 01:12:25.038 INFO (MainThread) [music_assistant.cache] Initializing cache controller...
2024-02-27 01:12:25.069 INFO (MainThread) [music_assistant.webserver] Starting server on  0.0.0.0:8095 - base url: http://192.168.178.130:8095
2024-02-27 01:12:25.137 INFO (MainThread) [music_assistant.music] Using a sync interval of 180 minutes.
2024-02-27 01:12:25.204 INFO (MainThread) [music_assistant.streams] Detected ffmpeg version 5.1.4 with libsoxr support
2024-02-27 01:12:25.204 INFO (MainThread) [music_assistant.streams] Starting server on  0.0.0.0:8096 - base url: http://192.168.178.130:8096
2024-02-27 01:12:25.507 INFO (MainThread) [music_assistant] Loaded player provider Universal Group Player
2024-02-27 01:12:25.507 INFO (MainThread) [music_assistant] Loaded music provider URL
2024-02-27 01:12:25.507 INFO (MainThread) [music_assistant] Loaded metadata provider MusicBrainz Metadata provider

**Reply by OzGav** [reply]
Have you changed any of the MA settings related to the webserver or streamserver?

Also things have evolved since your inital post. Can you confirm exactly what the problem is now.

**Reply by collateral87** [reply]
![grafik](https://github.com/music-assistant/hass-music-assistant/assets/97076257/6cee732e-73ad-4042-9c49-7a1374b0774e)
![grafik](https://github.com/music-assistant/hass-music-assistant/assets/97076257/8200f1c7-3655-4041-86f9-fc1ddaf63ee7)


I performed a reinstallation of the addon (server) during beta 102 or 103. Since then, I have not made any changes to the settings.

The problem still persists that I can set up the Home Assistant plugin and then the Home Assistant music provider, but as soon as I select any player that I want to import to MA, I only get this log entry:

![grafik](https://github.com/music-assistant/hass-music-assistant/assets/97076257/8413afc6-8a10-4e1f-83f2-31cab8a6b9a1)
V
V
V
![grafik](https://github.com/music-assistant/hass-music-assistant/assets/97076257/f270ab7b-eeb5-4e49-aba7-9e31d79362a1)

and in the player provider list, the new player isnt there.

(Thank you for your help so far)

**Reply by OzGav** [reply]
OK you must have exposed the webserver port though as that isn't on by defauilt?

**Reply by OzGav** [reply]
The only non-standrd thing you are doing here is running HA in a VM. I suspect there is something going wrong there.

**Reply by collateral87** [reply]
Yes, i tested with and without expose.

**Reply by OzGav** [reply]
Looking at Discord people running UNraid had issues when (1) they had port conflicts with other things running on the network; (2) insufficient resources in the VM; (3) not enabling HOST networking mode; (4) other VM misconfiguration

People are running MA successfully on unraid by the looks of it although some have separately installed MA in docker

**Reply by collateral87** [reply, marked as answer]
![Screenshot_20240227_024024_Samsung Internet.jpg](https://github.com/music-assistant/hass-music-assistant/assets/97076257/c07ab699-fcdb-44ff-b542-fa5725de15f8)

I followed the official guide on home-assistant.io.

But i will check all my network settings again..
